### ISSUE TITLE
Documents keeps coming back after deleting them

### ISSUE THREAD

**Issue by arinef** [alexcorvi/apexo#125]
**Describe the bug**
I noticed the documents that I delete will come back after I "Reload" the page or logged out then log in again

**To Reproduce**
1. Login to Apexo
2. go to Patients and create a new one but type multiple characters so it takes multiple Rev
3. Now delete that record
4. press "F5" to reload or logout then login again

You will notice the same record will appear again (but the older rev), The record will be deleted from "CouchDB", but it will not from local "PouchDB"

**Screenshots**
**Before Delete**
![delete doc](https://user-images.githubusercontent.com/51496889/59150819-28dbc480-8a32-11e9-8c57-383721d108c9.png)

**After Delete**
![after delete](https://user-images.githubusercontent.com/51496889/59150822-32fdc300-8a32-11e9-9f51-1290639c3e0c.PNG)

**After Reload**
![after refresh](https://user-images.githubusercontent.com/51496889/59150826-3db85800-8a32-11e9-99a4-1a81ba78baeb.PNG)

**Additional context**
Also, I felt some delay when I try to input very fast at the name field maybe it's due to syncing every character to remote DB, why don't we use a simple button to trigger the whole change one time like "save" button

**Comment by alexcorvi** [comment]
This was one of the harder issues to solve... simply because it was kind of tricky to reproduce.

It happened because when a document gets deleted from pouchDB, pouchDB will release an event that can be interpreted (by my implementation) that a document (which is a previous version, mostly a conflict) has been added. And my implementation, by design, can never distinguish between a truly new document and such case, unless, I write some sort of persistent list of deleted documents.

So documents kept being deleted then re-added.

In addition to this bug, I noticed few other shortcomings about my implementation, so I trashed the whole thing. And I wrote [pouchx](https://github.com/alexcorvi/pouchx) a re-usable implementation that will always keep pouchDB and mobx in sync. Using it, everything worked like it should.
